File: sdks/go/models/operational_webhook.go
package models

import "time"

type OperationalWebhookEndpointOut struct {
	Id          string    `json:"id"`
	CustomerId  string    `json:"customerId"`
	Url         string    `json:"url"`
	Description *string   `json:"description,omitempty"`
	IsActive    bool      `json:"isActive"`
	EventTypes  []string  `json:"eventTypes,omitempty"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

type OperationalWebhookEndpointIn struct {
	Url         string   `json:"url"`
	Description *string  `json:"description,omitempty"`
	IsActive    *bool    `json:"isActive,omitempty"`
	EventTypes  []string `json:"eventTypes,omitempty"`
}

type OperationalWebhookDeliveryOut struct {
	Id             string                 `json:"id"`
	EndpointId     string                 `json:"endpointId"`
	EventType      string                 `json:"eventType"`
	Payload        map[string]interface{} `json:"payload"`
	ResponseStatus *int16                 `json:"responseStatus,omitempty"`
	AttemptCount   int16                  `json:"attemptCount"`
	Status         string                 `json:"status"`
	CreatedAt      time.Time              `json:"createdAt"`
	DeliveredAt    *time.Time             `json:"deliveredAt,omitempty"`
}

File: sdks/python/hooksniff/models/operational_webhook.py
import typing as t
from datetime import datetime
from .common import BaseModel

class OperationalWebhookEndpointOut(BaseModel):
    id: str
    customer_id: str
    url: str
    description: t.Optional[str] = None
    is_active: bool
    event_types: t.Optional[t.List[str]] = None
    created_at: datetime
    updated_at: datetime

class OperationalWebhookEndpointIn(BaseModel):
    url: str
    description: t.Optional[str] = None
    is_active: t.Optional[bool] = None
    event_types: t.Optional[t.List[str]] = None

class OperationalWebhookDeliveryOut(BaseModel):
    id: str
    endpoint_id: str
    event_type: str
    payload: t.Dict[str, t.Any]
    response_status: t.Optional[int] = None
    attempt_count: int
    status: str
    created_at: datetime
    delivered_at: t.Optional[datetime] = None
